### ISSUE TITLE
dateutil.tz.tzlocal() has no localize attribute

### ISSUE THREAD

**Issue by JonasT** [dateutil/dateutil#790]
I was under the impression that `dateutil.tz.tzlocal()` is supposed to behave like any regular pytz time zone, e.g. `pytz.utc`- but the pytz time zone objects have a `localize` function to apply to naive datetime objects, and `dateutil.tz.tzlocal()` doesn't. Is this a bug or am I missing something?

**Comment by JonasT** [comment]
A lot of internet examples combine pytz and dateutil. Also, if dateutil doesn't *need* localize, why not offer it anyway for compatibility and make it do exactly what yo usaid? Would make it easier for people to transition over

**Comment by pganssle** [comment]
@JonasT `pytz`'s `localize` does something different and something that is *not* supported by `dateutil`. If you read my link you will see why using `pytz` at all is probably a mistake.

If people want to support both `pytz` and `dateutil.tz`, they can use:

```python
def localize(dt, tz):
    if hasattr(tz, 'localize'):
        return tz.localize(dt)
    else:
        return dt.replace(tzinfo=tz)
```

Though frankly I think it's usually unnecessary to actively support `pytz` in this way.

**Comment by pganssle** [comment]
> It isn't obvious to newcomers that .replace(tzinfo=xxx) does the same as localize for pytz, and the internet is full with pytz examples. 

That's because it doesn't.  They do two different things, and `pytz` also has a `normalize` method that is unnecessary in `dateutil`. The function I wrote just *looks* like it does the same thing, but all it does is create an aware datetime. I'm not interested in supporting `pytz`'s non-standard API, nor making internet examples that  "just work" with `pytz` work with `dateutil`.

One thing to note is that while this is not guaranteed (because neither project makes any guarantees with respect to the other), `dateutil.tz.gettz(str(TZ))` will work for any `pytz` time zone generated by `hypothesis.extras.pytz.timezones` (so, basically, all the `pytz` zones), so you can always just convert your `pytz` time zones "at the border" to `dateutil.tz` zones, then use the standard Python time zone interface.

**Comment by JonasT** [comment]
Right. I understand your position and you make very valid points. But the truth is, none of this is very relevant for newcomers who just want proper time-zone aware objects, and adding similar functions would just make jumping in easier.

But I completely understand saying "I don't want to clutter up the API with a nonsense function just to lower the entry barrier a little" :smile: I'm a bit biased, because for me this would have saved an hour of my time, so that's why I suggested it.
